### PR TITLE
PBI-03: Interactive Quiz app Skeleton

### PR DESCRIPTION
**Pull request by rashadaziz**
image
image

### CODE REVIEW

File: components/features/Dashboard/Applications/InteractiveQuiz/AssignmentTimer/AssignmentTimer.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect, useRef, useMemo } from "react";
+import styles from "./AssignmentTimer.module.css";
+
+interface Props extends React.PropsWithChildren {
+  durationInMinutes: number;
+  onTimerEnd?: () => void;
+}
+
+const timeFormatter = (totalSeconds: number) => {
```
Comment (whenwe177) [inline components/features/Dashboard/Applications/InteractiveQuiz/AssignmentTimer/AssignmentTimer.tsx:9]: Hello, rashad! Maybe in the future, if possible, we can refactor the time formatter into its own utils function, so we can use it somewhere else in our codebase. 😁

Comment (rashadaziz) [reply components/features/Dashboard/Applications/InteractiveQuiz/AssignmentTimer/AssignmentTimer.tsx:9]: Hi Alex good catch, yes I do plan on extracting this to a separate util function in the future. Thanks for the feedback :D

File: components/features/Dashboard/Applications/InteractiveQuiz/AssignmentTimer/AssignmentTimer.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  return `${hours}:${minutes}:${seconds}`;
+};
+
+const AssignmentTimer: React.FC<Props> = ({
+  durationInMinutes,
+  onTimerEnd,
+}) => {
+  const timerRef = useRef<NodeJS.Timer>();
+  const [timeRemaining, setTimeRemaining] = useState(() => durationInMinutes * 60);
```
Comment (whenwe177) [inline components/features/Dashboard/Applications/InteractiveQuiz/AssignmentTimer/AssignmentTimer.tsx:26]: Hello, rashad. I'm a bit curious, if I may ask, the time remaining unit used here is seconds, cmiiw? Let me know 😁

Comment (rashadaziz) [reply components/features/Dashboard/Applications/InteractiveQuiz/AssignmentTimer/AssignmentTimer.tsx:26]: Hello Alex, yes it is in seconds. From consulting with the backend team, they will send the duration in minutes:D

**Pull request merged**